### DISCUSSION TITLE
How to understand time_offsets

### DISCUSSION THREAD

**Discussion by qweasdf1354** [AdaptiveMotorControlLab/CEBRA#107] [Q&A] [answered]
Hi, i am new to CEBRA, and I just reading the doc and I do not understand the time_offsets in setting cebra_model.
the definition is following 
This corresponds to the distance (in time) between positive pairs and informs the algorithm about the time-scale of interest.

is the neural activity windows? or about the behaviour data? what is the default 10 means? the behaviour of mice i observed may delay the neural activity in about 10 second. is that related time_offsets setting?

Thanks a lot!

**Comment by MMathisLab** [comment, marked as answer]
Welcome! 👋 the offset can be thought of as a binning set, or "receptive field" in time; this would be in the neural space (if neurons are your time-series input). So for example in the hippocampus demo we use offset 10, as is 10 time bins of neural spikes that was recorded at 120hz.  This is a parameter we recommend finding what works best via our grid search. Hope that helps!

**Comment by qweasdf1354** [comment]
Thanks for your reply!

**Comment by sofiagilardini** [comment]
Hello! Thank you very much for your work. I have been using it in various projects and I am finding it very useful. I am, however, confused by the difference between the time_offsets that is defined as a hyperparameter, and the offset that is defined in the `get_offset()` method of the model architecture (e.g `return cebra.data.Offset(5, 5)`). Specifically with respect to the answer to this question, I do not understand how time_offsets can be thought of as the receptive field, as I thought that the receptive field was defined by the offset in the model architecture. 

In the example below, I can see from the [model definition](https://github.com/AdaptiveMotorControlLab/CEBRA/blob/31460de34c2fbc49dfb7b02d0ae9d45605beeeb3/cebra/models/model.py#L702C1-L704C41) that the receptive field of 'offset36-model' is (18, 18). How then should I think of the time_offset hyperparameter (set as 25 in this example)? 

```

                    cebra_model = CEBRA(
                        model_architecture = 'offset36-model',
                        batch_size= 256,
                        temperature_mode='auto',
                        learning_rate = 0.0001,
                        max_iterations = iterations,
                        time_offsets = 25, 
                        output_dimension = dim, 
                        device = "cuda_if_available",
                        verbose = True,
                        conditional='time_delta',
                        distance = 'cosine' 
                    )   

```

**Reply by sofiagilardini** [reply]
Hi @MMathisLab, 
Apologies for bringing this up again. I was wondering if you could please clarify my question above as the previous answer was not completely clear to me. I'd greatly appreciate it! 

Thank you :)

**Reply by MMathisLab** [reply]
Hey! Sorry this slipped off my radar! So the model offset is like binning; the time offset is the time window for samples. If you have model offset1, and time1, that is effectively not time contrastive learning (and akin to our benchmarking tSNE, umap, etc in figure 1). If I have a model offset 10, it has a receptive field of 5 before, 5 after. Then, if I use time offset 10, it would have a window of 10 time inputs..., etc. Does that help?

**Reply by sofiagilardini** [reply]
That's very helpful, thank you!
